pylearn2/datasets/hdf5.py

```diff
@@ -233,7 +239,26 @@ def next(self):
             try:
                 this_data = data[next_index]
             except TypeError:
-                this_data = data[next_index, :]
+                # FB: Why this try..except is there? I think this is useless.
+                # Do not hide the original if we can't fall back.
+                # FV: This is triggered if the shape of next_index is
+                # incompatible with the shape of the dataset. See for an
+                # example test_hdf5_topo_view(), where where i
+                # next.index.shape = (10,) and data is 'data': <HDF5
+                # dataset "y": shape (10, 3), type "<f8">
+                # I think it would be better to explicitly check if
+                # next_index.shape is incompatible with data.shape, for
+                # instance checking if next_index.ndim == data.ndim
+                if data.ndim > 1:
+                    this_data = data[next_index, :]
+                else:
+                    raise
+            # Check if the dataset data is a vector and transform it into a
+            # one-column matrix. This is needed to automatically convert the
+            # shape of the data later (in the format_as method of the
+            # Space.)
+            if this_data.ndim == 1:
+                this_data = this_data[:, np.newaxis]
```

Comment:
This is too general. Some spaces may require data of dimensionality 1 in the future, and this would not work.